packages/ozone/src/api/setting/upsertOption.ts

```diff
@@ -0,0 +1,136 @@
+import { AuthRequiredError } from '@atproto/xrpc-server'
+import { Server } from '../../lexicon'
+import AppContext from '../../context'
+import { AdminTokenOutput, ModeratorOutput } from '../../auth-verifier'
+import { SettingService } from '../../setting/service'
+import { Member } from '../../db/schema/member'
+import { ToolsOzoneTeamDefs } from '@atproto/api'
+
+export default function (server: Server, ctx: AppContext) {
+  server.tools.ozone.setting.upsertOption({
+    auth: ctx.authVerifier.modOrAdminToken,
+    handler: async ({ input, auth }) => {
+      const access = auth.credentials
+      const db = ctx.db
+      const { key, value, description, managerRole, scope } = input.body
+      const serviceDid = ctx.cfg.service.did
+      let ownerDid = serviceDid
+
+      if (scope === 'personal' && access.type !== 'moderator') {
+        throw new AuthRequiredError(
+          'Must use moderator auth to create or update a personal setting',
+        )
+      }
+
+      // if the caller is using moderator auth and storing personal setting
+      // use the caller's DID as the owner
+      if (scope === 'personal' && access.type === 'moderator') {
+        ownerDid = access.iss
+      }
+
+      const now = new Date()
+      const baseOption = {
+        key,
+        value,
+        did: ownerDid,
+        createdBy: ownerDid,
+        lastUpdatedBy: ownerDid,
+        description: description || '',
+        createdAt: now,
+        updatedAt: now,
+      }
+
+      const settingService = ctx.settingService(db)
+      if (scope === 'personal') {
+        await settingService.upsert({
+          ...baseOption,
+          scope: 'personal',
+          managerRole: null,
+        })
+      } else {
+        const manageableRoles = getRolesForInstanceOption(access)
+        const existingSetting = await settingService.query({
+          scope: 'instance',
+          keys: [key],
+          limit: 1,
+        })
```

Comment:
Could `getExistingSetting()` be used here too?

Reply:
that's why i refactored it into the function but forgot to replace in both places.